R/kernel.r

```diff
@@ -300,20 +314,30 @@ initialize = function(connection_file) {
 
 run = function() {
     while (TRUE) {
+        debug("main loop: beginning")
         zmq.poll(
             c(sockets$hb, sockets$shell, sockets$control),
             rep(.pbd_env$ZMQ.PO$POLLIN, 3))
-        
-        if(bitwAnd(zmq.poll.get.revents(1), .pbd_env$ZMQ.PO$POLLIN))
```

Comment:
As long as a new poll does not remove an already stored message, a `if ... Else if ... Else ...` should also work... Will try later...